CustomKernelInstall: handle unexpected kernel upgrade in Ubuntu

chvalean opened the pull request:
Fixes #311 

Before:
```
root@vm:/home/user# apt remove linux-cloud-tools-common
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  linux-azure-headers-4.15.0-1046 linux-azure-tools-4.15.0-1046 linux-headers-4.15.0-1046-azure linux-headers-azure linux-image-4.15.0-1046-azure linux-image-azure linux-modules-4.15.0-1046-azure
  linux-tools-4.15.0-1046-azure linux-tools-azure
Suggested packages:
  fdutils linux-azure-doc-4.15.0 | linux-azure-source-4.15.0 linux-azure-tools
The following packages will be REMOVED:
  linux-azure linux-azure-cloud-tools-4.15.0-1045 linux-cloud-tools-4.15.0-1045-azure linux-cloud-tools-azure linux-cloud-tools-common
The following NEW packages will be installed:
  linux-azure-headers-4.15.0-1046 linux-azure-tools-4.15.0-1046 linux-headers-4.15.0-1046-azure linux-image-4.15.0-1046-azure linux-modules-4.15.0-1046-azure linux-tools-4.15.0-1046-azure
The following packages will be upgraded:
  linux-headers-azure linux-image-azure linux-tools-azure
3 upgraded, 6 newly installed, 5 to remove and 23 not upgraded.
```


After:
```
root@vm:/home/user# dpkg --remove --force-depends linux-cloud-tools-common
dpkg: linux-cloud-tools-common: dependency problems, but removing anyway as you requested:
 linux-azure-cloud-tools-4.15.0-1045 depends on linux-cloud-tools-common.

(Reading database ... 53456 files and directories currently installed.)
Removing linux-cloud-tools-common (4.4.0-148.174) ...
```

lizzha commented on Libraries/Azure.psm1:
```diff
@@ -452,14 +452,18 @@ Function CreateAllResourceGroupDeployments($setupType, $xmlConfig, $Distro, [str
-                {
-                    $groupName = "ICA-RG-" + $RG.Tag + "-" + $Distro + "-" + "$shortRandomWord-" + "$curtime"
+                if ( $RGLock ) {
+                    $PreambleString = "ICB-RG-"
```
Instead of this hardcoded string as RG prefix, I think it's better to enable LISAv2 to accept prefix from parameter.
e.g. Add a parameter named like "-prefix", if it's provided, the test use it as RG prefix, otherwise it use ICA as RG prefix

ader1990 replied:
I agree this change has zero plus for lisav2 framework, it is a hack to override some actual environment problem that you have only on Azure. A ResurceGroupPrefix parameter should be more generic.

iamshital replied:
Agree with @ader1990 and @lizzha . Add a optional parameter "ResourceGroupPrefix".

juhlee-microsoft replied:
@iamshital @ader1990 @lizzha 
I am not sure your feedback is correct because the originally it is hardcoded 'ICA-RG-' in LISAv2. The RG cleanup tool uses this preamble string as the filter. If we open this RG Premable string with ResourceGroupPrefix, then the RG cleanup tool got useless or complicated work. I think this is the only way we can manage RG with Cleanup tool schedule.

lizzha replied:
In most cases, the default prefix should be used. Only if you need to opt-out from the cleanup tool, you need to run the test with customized prefix. When using customized prefix, it's still required that we use a prefix like alias which helps to identify the RG owner.

juhlee-microsoft replied:
RG owner should be found by tagging feature recently added. I am not sure still customed RG name is good choice. Let me discuss with the team tomorrow

lizzha commented on Run-LisaV2.ps1:
```diff
@@ -371,8 +372,8 @@ try {
 	if ( $TiPCluster) { $command += " -TiPCluster $TiPCluster" }
-	if ($UseManagedDisks) {	$command += " -UseManagedDisks" }
-	if ($XMLSecretFile) { $command += " -XMLSecretFile '$XMLSecretFile'" }
+	if ( $UseManagedDisks) {	$command += " -UseManagedDisks" }
```
Is this indent change expected?

ader1990 replied:
don t change stuff that is unrelated to the code commit or purpose

juhlee-microsoft replied:
I did not mean too. It changed to tab though. Fixed

juhlee-microsoft replied:
fixed 423bb3e3c5cb4d52cb4b950168e7e073ff5537e6

ader1990 commented on Run-LisaV2.ps1:
```diff
@@ -105,7 +106,7 @@ try {
 	if ( $WorkingDirectory.Length -gt $MaxDirLength) {
 		$OriginalWorkingDirectory = $WorkingDirectory
-		Write-Output "Current working directory '$WorkingDirectory' length is greather than $MaxDirLength."
+		Write-Output "Current working directory '$WorkingDirectory' length is greater than $MaxDirLength."
```
don t change stuff that is unrelated to the code commit or purpose

juhlee-microsoft replied:
Will open a new PR for this single typo. This has not looked better Company branding from the beginning of LISAv2. It was not good to show a typo in the external presentation.

juhlee-microsoft replied:
reverted 423bb3e3c5cb4d52cb4b950168e7e073ff5537e6

juhlee-microsoft replied:
#333

chvalean committed:
CustomKernelInstall: handle unexpected kernel upgrade in Ubuntu


Before:
root@vm:/home/user# apt remove linux-cloud-tools-common
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  linux-azure-headers-4.15.0-1046 linux-azure-tools-4.15.0-1046 linux-headers-4.15.0-1046-azure linux-headers-azure linux-image-4.15.0-1046-azure linux-image-azure linux-modules-4.15.0-1046-azure
  linux-tools-4.15.0-1046-azure linux-tools-azure
Suggested packages:
  fdutils linux-azure-doc-4.15.0 | linux-azure-source-4.15.0 linux-azure-tools
The following packages will be REMOVED:
  linux-azure linux-azure-cloud-tools-4.15.0-1045 linux-cloud-tools-4.15.0-1045-azure linux-cloud-tools-azure linux-cloud-tools-common
The following NEW packages will be installed:
  linux-azure-headers-4.15.0-1046 linux-azure-tools-4.15.0-1046 linux-headers-4.15.0-1046-azure linux-image-4.15.0-1046-azure linux-modules-4.15.0-1046-azure linux-tools-4.15.0-1046-azure
The following packages will be upgraded:
  linux-headers-azure linux-image-azure linux-tools-azure
3 upgraded, 6 newly installed, 5 to remove and 23 not upgraded.


After:
root@vm:/home/user# dpkg --remove --force-depends linux-cloud-tools-common
dpkg: linux-cloud-tools-common: dependency problems, but removing anyway as you requested:
 linux-azure-cloud-tools-4.15.0-1045 depends on linux-cloud-tools-common.

(Reading database ... 53456 files and directories currently installed.)
Removing linux-cloud-tools-common (4.4.0-148.174) ...

chvalean committed:
Update customKernelInstall.sh
